scripts/e2e-experimental.sh

```diff
@@ -0,0 +1,284 @@
+         echo "ERROR: Must specify a feature scope flag when invoking this script."
+         usage
+         exit 1
+    fi
+}
+
+########################################
+# This function initializes the directory
+# used during end-to-end testing. This function
+# also creates the directory structure which
+# instructlab expects to see at the time of execution.
+# Arguments:
+#   None
+# Globals:
+#   DATA_HOME
+#   CACHE_HOME
+#   CONFIG_HOME
+#   E2E_TEST_DIR
+#   HOME
+# Outputs:
+#   None
+########################################
+
+init_e2e_tests() {
+    E2E_TEST_DIR=$(mktemp -d)
+    export HOME="${E2E_TEST_DIR}"  # update the HOME directory used to resolve paths
+
+    CONFIG_HOME=$(python -c 'import platformdirs; print(platformdirs.user_config_dir())')
+    DATA_HOME=$(python -c 'import platformdirs; print(platformdirs.user_data_dir())')
+    CACHE_HOME=$(python -c 'import platformdirs; print(platformdirs.user_cache_dir())')
+    # ensure that our mock e2e dirs exist
+    for dir in "${CONFIG_HOME}" "${DATA_HOME}" "${CACHE_HOME}"; do
+        mkdir -p "${dir}"
+    done
+
+    E2E_LOG_DIR="${HOME}/log"
+    mkdir -p "${E2E_LOG_DIR}"
+}
+
+step() {
+    echo -e "$BOLD$* - $(date)$NC"
+}
+
+task() {
+    echo -e "$BOLD------------------------------------------------------$NC"
+    step "$@"
+}
+
+check_disk() {
+   task Check disk
+   df -h
+}
+
+test_smoke() {
+    task InstructLab smoke test
+    ilab | grep --color 'Usage: ilab'
+    task InstructLab smoke test Complete
+}
+
+test_system_info() {
+    task Output system info
+    ilab system info
+    task System info Complete
+}
+
+test_init() {
+    task Initializing InstructLab
+    step Setting small-size system profile
+    ilab config init --non-interactive --profile "${SCRIPTDIR}/test-data/profile-t4-x1.yaml"
+    task InstructLab initialization Complete
+}
+
+test_config_show() {
+    task Output active config
+    ilab config show
+    task Config show Complete
+}
+
+test_download() {
+    task Download models
+    step Downloading granite-7b-lab model for chat
+    ilab model download --repository ${GRANITE_7B_MODEL}
+    step Downloading the granite-embedding-125m-english model for embedding
+    ilab model download --repository ${GRANITE_EMBEDDING_MODEL}
+    task Downloading models Complete
+}
+
+test_list() {
+    task List the Downloaded Models
+    ilab model list | grep ${GRANITE_7B_MODEL}
+    ilab model list | grep ${GRANITE_EMBEDDING_MODEL}
+    task Model Listing Complete
+}
+
+test_taxonomy() {
+    task Update the taxonomy
+    step Add knowledge to the taxonomy
+    mkdir -p "$DATA_HOME"/instructlab/taxonomy/knowledge/phoenix/overview/e2e-phoenix
+    cp "$SCRIPTDIR"/test-data/knowledge/e2e-qna-knowledge-phoenix.yaml "$DATA_HOME"/instructlab/taxonomy/knowledge/phoenix/overview/e2e-phoenix/qna.yaml
+
+    step Verify the taxonomy
+    ilab taxonomy diff
+
+    task Taxonomy updates and verification Complete
+}
+
+test_convert() {
```

Comment:
if you want to convert taxonomy knowledge files:
`ilab rag convert`
if you want to convert user docs:
`ilab rag convert --input-dir "$SCRIPTDIR"/test-data/raw_documents`

Probably we can test both flows, since we have the taxonomy knowledge files already in. Just put the outputs in different folders with `--output-dir ....` option

Reply:
Good call, done.